### ISSUE TITLE
Error during podElection with undefined podIP

### ISSUE THREAD

**Issue by EIrwin** [cvallance/mongo-k8s-sidecar#59]
We have been trying to identify some intermittent issues with our MongoDB cluster running on stateful sets using this sidecar. In our logs leading up to some of our issues, we are seeing the following logs.

I am not even sure where to begin to identify what could be causing this, and I haven't seen any issues related to this.

```
TypeError: Cannot read property 'split' of undefined at Object.toInt [as toLong] 
(/opt/cvallance/mongo-k8s-sidecar/node_modules/ip/lib/ip.js:364:5) at /opt/cvallance/mongo-k8s-
sidecar/src/lib/worker.js:215:21 at Array.sort (native) at podElection (/opt/cvallance/mongo-k8s-
sidecar/src/lib/worker.js:213:8) at inReplicaSet (/opt/cvallance/mongo-k8s-
sidecar/src/lib/worker.js:124:25) at /opt/cvallance/mongo-k8s-sidecar/src/lib/worker.js:85:7 at 
/opt/cvallance/mongo-k8s-sidecar/src/lib/mongo.js:68:12 at /opt/cvallance/mongo-k8s-
sidecar/node_modules/mongodb/lib/admin.js:77:31 at handleCallback (/opt/cvallance/mongo-k8s-
sidecar/node_modules/mongodb/lib/utils.js:120:56) at /opt/cvallance/mongo-k8s-
sidecar/node_modules/mongodb/lib/db.js:995:7

```

This points at the `podElection(...)` function defined at https://github.com/cvallance/mongo-k8s-sidecar/blob/master/src/lib/worker.js#L214.

**Comment by EIrwin** [comment]
Not sure if anybody has had this occur, or has even looked at this, but my guess is that this is directly related to failures with `kube-proxy` init container stuck in `Waiting` phase. Once we identified this had happened, and killed the node that the proxy was on, we haven't had this occur.

[https://github.com/kubernetes/kubernetes/issues/52462](url)

**Comment by cvallance** [comment]
Thanks for the info! Closing issue due to k8s bug (which looks to be fixed in 1.7).
